packages/ui-keyring/src/Keyring.ts

```diff
@@ -82,6 +83,24 @@ export class Keyring extends Base implements KeyringStruct {
     return pair.toJson(password);
   }
 
+  public async backupAccounts (addresses: string[]): Promise<KeyringPairs$Json> {
+    const accountPromises = addresses.map((address) => {
+      return new Promise<KeyringJson>((resolve) => {
+        this._store.get(accountKey(address), resolve);
+      });
+    });
+
+    const accounts = await Promise.all(accountPromises);
+
+    return {
+      ...jsonEncrypt(stringToU8a(JSON.stringify(accounts)), ['batch-pkcs8']),
+      metas: accounts.map((account) => ({
```

Comment:
Looks good. I would hoever suggest something a bit closer to the singular JSON format -

```js
addresses: accounts.map(({ address }) => address),
meta: accounts.map(({ meta }) => meta)
```

This is slightly less optimal since we def. do duplicate the metadata but it is generally not massive. Additionally it has the benefits of pulling through the `genesisHash` (showing the chain) as well as the tags (not used in the extension, but valuable elsewhere)

Reply:
Great, that is def. better than my suggestion.